porch/func/server/server.go

```diff
@@ -19,17 +19,26 @@ import (
 	"fmt"
 	"net"
 	"os"
+	"strings"
 
 	pb "github.com/GoogleContainerTools/kpt/porch/func/evaluator"
 	"github.com/GoogleContainerTools/kpt/porch/func/internal"
 	"google.golang.org/grpc"
 	"k8s.io/klog/v2"
 )
 
+const (
+	execRuntime = "exec"
+	podRuntime  = "pod"
+)
+
 var (
-	port      = flag.Int("port", 9445, "The server port")
-	functions = flag.String("functions", "./functions", "Path to cached functions.")
-	config    = flag.String("config", "./config.yaml", "Path to the config file.")
+	port               = flag.Int("port", 9445, "The server port")
+	functions          = flag.String("functions", "./functions", "Path to cached functions.")
+	config             = flag.String("config", "./config.yaml", "Path to the config file.")
+	podNamespace       = flag.String("pod-namespace", "porch-fn-system", "Namespace to run KRM functions pods.")
+	wrapperServerImage = flag.String("wrapper-server-image", "", "Image name of the wrapper server.")
+	disableRuntimes    = flag.String("disable-runtimes", "", fmt.Sprintf("The runtime(s) to disable. Multiple runtimes should separated by `,`. Available runtimes: `%v`, `%v`.", execRuntime, podRuntime))
```

Comment:
I'm not using the magic in [`IsUsingDevPorch()`](https://github.com/GoogleContainerTools/kpt/blob/6c027575f1291621cd79cdff7a8cf32823c09437/porch/apiserver/pkg/e2e/suite.go#L140-L153).
Instead I provide the flag to explicitly disable some runtimes in the function runner.
It's easier to find out which runtimes are disabled.

Reply:
So in a local environment the function runtime will be only 'exec'.
Now there are 2 options:
1) the e2e test doesn't use any functions that require the `pod` runtime. In this case my question is - how are we testing the `pod` runtime functionality in e2e tests?
2) the e2e test does use function(s) that require the `pod` runtime. In this case, my question is - how does the test not fail when running in local environment?

I haven't been able to find where the test checks "if pod runtime is available, use some functions that require it, otherwise, use only simpler configs that don't require it"